### PR TITLE
add device id and provisioning scripts

### PR DESCRIPTION
**Pull request by jayteemo**
add script to use provisioning API.
add script to automate claiming and provisioning.
IRIS-6795

### CODE REVIEW

**Comment by jayteemo** [comment]
Example usage:
`python3 claim_and_provision_device.py --api_key $API_KEY --stage dev --ca=./ca.pem --ca_key=ca_prv_key.pem --verbose`

provide your api key and the PEM files for _your_ CA cert and it's private key.
the script requires that the device is running the nrf provisioning sample... you can provide a hex file and it will flash the device.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py]: You need to rebase and switch to using `ca_certs.py` I think.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py:447]: If the device is already claimed, what happens?

Should we check for an error and abort here?

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py]: Maybe add an argument to the CLI to unclaim first.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py]: How about, if this is chosen on the cli, first flash the at client, install the CA certs, then flash the provisioning client?

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py:521]: yes, and CoAP if requested

File: python/modem-firmware-1.3+/nrf_cloud_diap.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/nrf_cloud_diap.py]: 2023

File: python/modem-firmware-1.3+/nrf_cloud_provision.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/nrf_cloud_provision.py:113]: Seems like this could allow a non-Nordic person to provision a device to dev or beta.  Maybe not a new problem, but perhaps a security risk?

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/claim_and_provision_device.py:266]: Maybe a future improvement, but, it would be good to be able to do this without USB too.  For example, the customer provides a csv file containing one or more attestation tokens.  The script would assume the device somewhere out there in the world is on and already has the provisioning client (or a customer application that includes the provisioning library) installed as well as the proper provisioning root CA.  The script would just interact with the provisioning REST API.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:521]: i wonder if it would just be better to update `nrf_provisioning` to always add both CAs?
my update last week of the `ca_certs.py` was not quite correct... 
the **AWS CA** is used for _provisioning over http_ and the coap CA is used for _provisioning over coap_ (which i didn't realize was implemented).

currently, `nrf_provisioning` writes the cert for the configured protocol:
```
#if defined(CONFIG_NRF_PROVISIONING_HTTP)
		/* Certificate for nRF HTTP Provisioning */
		const uint8_t *cert = cert_amazon_root_ca1_pem;
		const size_t cert_size = cert_amazon_root_ca1_pem_size;
#else
		/* Certificate for nRF COAP Provisioning */
		const uint8_t *cert = cert_coap_root_ca_pem;
		const size_t cert_size = cert_coap_root_ca_pem_size;
#endif
```

created IRIS-6918

File: python/modem-firmware-1.3+/nrf_cloud_provision.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/nrf_cloud_provision.py:113]: if they have an API key for dev or beta, they can do whatever they want :)

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: IRIS-6918

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:266]: IRIS-6919

**Comment by jayteemo** [comment]
added a commit to cleanup the CA cert usage.
removed the `--prov` option from the installer script since it is not needed.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:521]: Yeah, I have thought about that too (just always doing both).

File: python/modem-firmware-1.3+/nrf_cloud_provision.py
Comment (plskeggs) [inline python/modem-firmware-1.3+/nrf_cloud_provision.py:34]: I just occurred to me that if we are going to add dev and beta support here, we might as well add ca certs for them too.  See my branch: https://github.com/plskeggs/utils/tree/feature-coap-all-stages

**Review by plskeggs — APPROVED** [review]

File: python/modem-firmware-1.3+/nrf_cloud_provision.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/nrf_cloud_provision.py:34]: they use a different CA for dev and beta?
what is the reason for that?

Comment (jayteemo) [reply python/modem-firmware-1.3+/nrf_cloud_provision.py:34]: IRIS-6923

Comment (plskeggs) [reply python/modem-firmware-1.3+/nrf_cloud_provision.py:34]: I did not ask.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:521]: ok, i will leave it up to you to coordinate such a change with the `nrf_provisioning` code owners 😺

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (SoorajRajasekharan) [inline python/modem-firmware-1.3+/claim_and_provision_device.py]: Can it install the CoAP CA certificate based on the stage? CoAP CA certificates are different for these three environments.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (SoorajRajasekharan) [inline python/modem-firmware-1.3+/claim_and_provision_device.py]: provision.csv file was not generated after secure provisioning.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (SoorajRajasekharan) [inline python/modem-firmware-1.3+/claim_and_provision_device.py:521]: After the "Client Certificate Command," the next step should be to include the "Finish command" in the script in order to change the device status to "provisioned" in nrfcloud and restart the device after provisioning. Finish command should be added as the last command to change the status from Awaiting to Provisioned.
image

Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:521]: ah i see, "finish" in the test plan.  i didn't realize that was a provisioning command 😆

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: i forgot to remove this parameter, it was carried over from the device credentials installer script.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py:521]: image
Updated

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: removed

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: CA cert installation has been implemented.
You will need to build your provisioning client FW with the following options:
```
CONFIG_NRF_PROVISIONING_RX_BUF_SZ=2048
CONFIG_SHELL_BACKEND_SERIAL_RX_RING_BUFFER_SIZE=2048
CONFIG_SHELL_CMD_BUFF_SIZE=2048
```

**Comment by jayteemo** [comment]
please re-review/test with the new commits.

File: python/modem-firmware-1.3+/claim_and_provision_device.py
Comment (plskeggs) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: Should these Kconfig changes be added officially to the provisioning client prj.conf?

Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: not my decision.
you could ask Juha, as he helped me get it working: https://nordicsemi.atlassian.net/browse/NCSDK-23596

Comment (plskeggs) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: OK, I was unaware these were not your changes but came from Juha.

Comment (jayteemo) [reply python/modem-firmware-1.3+/claim_and_provision_device.py]: the changes to these python scripts are from me based on comments from Juha in the linked ticket.

**Review by plskeggs — APPROVED** [review]

**Pull request merged**
